src/chart/gauge/GaugeView.ts

```diff
@@ -312,165 +324,296 @@ class GaugeView extends ChartView {
+                    r: r
+                },
+                // 是否相应鼠标事件
+                silent: progressSilent
+            });
+            isOverlap && (progress.z2 = maxVal - (data.get(valueDim, idx) as number) % maxVal);
+            return progress;
+        }
 
-                graphic.updateProps(pointer, {
-                    shape: {
-                        angle: linearMap(data.get(valueDim, newIdx) as number, valueExtent, angleExtent, true)
+        if (showProgress || showPointer) {
+            data.diff(oldData)
+                .add(function (idx) {
+                    if (showPointer) {
+                        const pointer = createPointer(idx, startAngle);
+                        graphic.initProps(pointer, {
+                            rotation: -(linearMap(data.get(valueDim, idx) as number, valueExtent, angleExtent, true) + Math.PI / 2)
+                        }, seriesModel);
+                        group.add(pointer);
+                        data.setItemGraphicEl(idx, pointer);
                     }
-                }, seriesModel);
-
-                group.add(pointer);
-                data.setItemGraphicEl(newIdx, pointer);
-            })
-            .remove(function (idx) {
-                const pointer = oldData.getItemGraphicEl(idx);
-                group.remove(pointer);
-            })
-            .execute();
-
-        data.eachItemGraphicEl(function (pointer: PointerPath, idx) {
-            const itemModel = data.getItemModel<GaugeDataItemOption>(idx);
-            const pointerModel = itemModel.getModel('pointer');
-            const emphasisModel = itemModel.getModel('emphasis');
-
-            pointer.setShape({
-                x: posInfo.cx,
-                y: posInfo.cy,
-                width: parsePercent(
-                    pointerModel.get('width'), posInfo.r
-                ),
-                r: parsePercent(pointerModel.get('length'), posInfo.r)
-            });
 
-            pointer.useStyle(itemModel.getModel('itemStyle').getItemStyle());
+                    if (showProgress) {
+                        const progress = createProgress(idx, startAngle) as graphic.Sector;
+                        const isClip = progressModel.get('clip');
+                        graphic.initProps(progress, {
+                            shape: {
+                                endAngle: linearMap(data.get(valueDim, idx) as number, valueExtent, angleExtent, isClip)
+                            }
+                        }, seriesModel);
+                        group.add(progress);
+                        progressList[idx] = progress;
+                    }
+                })
+                .update(function (newIdx, oldIdx) {
+                    if (showPointer) {
+                        const previousPointer = oldData.getItemGraphicEl(oldIdx) as PointerPath;
+                        const previousRotate = previousPointer.rotation;
+                        group.remove(previousPointer);
```

Comment:
Should check if `previousPointer` exists.  It's possible to have error when switching the `show` property from false to true.